pilot/pkg/model/push_context_test.go

```diff
@@ -1613,7 +1613,7 @@ func TestMergeGatewayServerPort(t *testing.T) {
 					ServicePort: &Port{Port: 8},
 				},
 			},
-			wantedPorts: map[uint32]struct{}{8: {}},
+			wantedPorts: map[uint32]struct{}{7: {}, 8: {}},
```

Comment:
what would happen if wehave a gw for port 7,8, and 9?

Reply:
None of the ports will be rewritten, and in this case gateway is guarantee to fail due to duplicated listeners.